src/main/java/ru/l1ttleO/boyfriend/commands/Remind.java

```diff
@@ -1,43 +1,79 @@
+                    if (listText.isEmpty())
+                        append.strip();
+                    if (listText.length() + append.length() > 2000) {
+                        channel.sendMessage(listText).queue();
+                        listText.setLength(0); // clear
+                        if (append.length() > 2000)
+                            append = append.substring(0, 1994) + "..." + append.substring(append.length() - 3, append.length());
+                    }
+                    listText.append(append);
+                }
+                channel.sendMessage(listText).queue();
+            }
+            return;
+        } else if (args.length < 3)
+            throw new WrongUsageException("Требуется указать текст напоминания!");
         if (author == null)
             throw new InvalidAuthorException();
-        duration = Utils.getDurationMultiplied(args[1]);
+        try {
+            duration = Utils.getDurationMultiplied(args[1]);
+        } catch (final @NotNull NumberFormatException e) {
+            throw new WrongUsageException("Неверно указана продолжительность!");
+        }
         if (duration < 1)
-            throw new WrongUsageException("Требуется указать продолжительность!", channel, this.getUsages());
-        text = " ```" + StringUtils.join(args, ' ', 2, args.length).replaceAll("```", "​`​`​`​") + " ```";
-        channel.sendMessage("Напоминание успешно установлено. Через %s будет отправлено данное сообщение:%s".formatted(Utils.getDurationText(duration, true), text)).queue();
-        final Thread thread = new Thread(() -> {
-            try {
-                Thread.sleep(duration * 1000L);
-                channel.sendMessage(author.getAsMention() + text).queue();
-            } catch (final InterruptedException e) {
-                Actions.sendNotification(event.getGuild(), "Прерван таймер напоминания для %s: %s".formatted(author.getAsMention(), text), false);
-            }
-        }, "Remind timer " + author.getId());
-        thread.start();
+            throw new WrongUsageException("Длительность должна быть больше нуля!");
+        text = Utils.wrap(StringUtils.join(args, ' ', 2, args.length));
+        channel.sendMessage("Напоминание успешно установлено. Через %s будет отправлено данное сообщение: %s".formatted(Utils.getDurationText(duration, true), text)).queue();
+
+        final DelayedRunnable runnable = new DelayedRunnable(REMINDERS_THREAD_GROUP, (final DelayedRunnable dr) -> {
+            channel.sendMessage(author.getAsMention() + " " + text).queue();
+            REMINDERS.getOrDefault(author.getIdLong(), new HashMap<>()).remove(dr);
+        }, "Remind timer " + author.getId(), duration * 1000L, (final DelayedRunnable dr) -> {
+            Actions.sendNotification(event.getGuild(), "Прерван таймер напоминания для %s: %s".formatted(author.getAsMention(), text), false);
+        });
```

Comment:
```suggestion
        }, "Remind timer " + author.getId(), duration * 1000L, (final DelayedRunnable dr) -> Actions.sendNotification(event.getGuild(), "Прерван таймер напоминания для %s: %s".formatted(author.getAsMention(), text), false));
```
Reason: `Statement lambda can be replaced with expression lambda`

Reply:
Maybe keep a first line break for readability?

Reply:
sure

Reply:
I think you misplaced a whitespace and broke indentation

Reply:
IntelliJ thinks it's indented correctly

Reply:
ok, whatever
maybe GitHub plays with me, idk

Reply:
see what I said?

Reply:
![image](https://user-images.githubusercontent.com/61277953/129330639-baf40a29-4ad1-4eeb-a8f3-d20a31023057.png)
did it not copy-paste properly?

Reply:
> edit: fixed, you can commit this suggestion now

no, I can't
![image](https://user-images.githubusercontent.com/53149450/129331763-04a3187f-5f13-45c6-bda4-d4afad264574.png)

Reply:
fixed